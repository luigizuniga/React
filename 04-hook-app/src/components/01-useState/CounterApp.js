import React, { useState, Fragment } from 'react';

export const CounterApp = () => {
    const [ normalState, setNormalState ] = useState(100);
    const [ state, setState ] = useState({
        counter_1: 0,
        counter_2: 2,
        counter_3: 9
    });

    const { counter_1, counter_2 } = state;

    return (
        <Fragment>
             <h1 className="bd-title mt-5" id="content">CounterApp Component</h1>
            <div className="row lign-items-center mt-2">
                <div className="col-md-6">
                    <label for="control_1" className="form-label"> COUNTER PROPERTY 1 </label>
                    <input type="text" className="form-control " placeholder="value"
                           aria-label="control_1" id="control_1" value={ counter_1 }/>
                </div>
                <div className="col-md-6">
                    <label for="control_2" className="form-label">COUNTER PROPERTY 2</label>
                    <input type="text" className="form-control " placeholder="value"
                           aria-label="control_2" id="control_2" value={ counter_2 }/>
                </div>
                <div className="col-md-6">
                    <label for="normalState" className="form-label">COUNTER NORMAL</label>
                    <input type="text" className="form-control " placeholder="value"
                          aria-label="normalState" id="normalState" value={ normalState }/>
                </div>
            </div>
            <hr/>
            <div className="d-grid gap-2 d-md-flex justify-content-lg-end">
                <button className="btn btn-primary me-md-2" type="button"
                         onClick={ () => {
                            setState({
                                ...state,
                                counter_1: counter_1 + 1,
                                counter_2: counter_2 + 2
                            });
                        }}
                > + : CP1 </button>
                <button className="btn btn-danger" type="button"
                         onClick={ () =>
                            setState({
                                ...state,
                                counter_1: counter_1 - 1,
                                counter_2: counter_2 - 2
                            })
                         }
                > - : CP2 </button>
                              <button className="btn btn-warning" type="button"
                         onClick={ () => setNormalState( normalState + 100 )
                         }
                > + 100 </button>
            </div>
        </Fragment>
    )
}
