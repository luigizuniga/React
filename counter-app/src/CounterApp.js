import React, { Fragment ,useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ count= 100 }) => {
    const [ counter, setCounter ] = useState(count); // []

    const handleAdd = () => setCounter((c) => c + 1);
    const handleSubtract = () => setCounter((c) => c - 1);
    const handleReset = () => setCounter( count );

    return (
        <Fragment>
            <h2>Bienvenido a CounterApp</h2>
            <p>{ counter }</p>
            <button onClick={ handleAdd }>+1</button>
            <button onClick={ handleReset }>Reset</button>
            <button onClick={ handleSubtract }>-1</button>
        </Fragment>
    );
}

CounterApp.propTypes = {
    count: PropTypes.number
}


export default CounterApp;