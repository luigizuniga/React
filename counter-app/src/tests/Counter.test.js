import React from 'react';
import { shallow } from 'enzyme';

import CounterApp from '../CounterApp';

describe('Pruebas en el <CounterApp/>', () => {
    // let wrapper = shallow( <CounterApp count={ 100 }/>);
    // beforeEach(()=>{
    //     wrapper = shallow( <CounterApp />);
    // })
    test('Debe de mostrar <CounterApp /> correctamente', () => {
        const wrapper = shallow(<CounterApp/>);
        console.log(wrapper.html())
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe de mostrat los elemtos html de cada boton', ()=>{
        const wrapper = shallow(<CounterApp />);

        const btnAdd = wrapper.find('button').at(0);
        const btnRes = wrapper.find('button').at(1);
        const btnSub = wrapper.find('button').at(2);

        console.log(btnAdd.html());
        console.log(btnRes.html());
        console.log(btnSub.html());
    })

    test('Debe de mostrar el valor por defecto de 100', () => {
        // Revision
        // Prueba unitaria sin resultado esperado
        const wrapper = shallow( <CounterApp/> );
        const counterText = wrapper.find('p').text().trim();
        console.log(counterText);
        expect( counterText ).toBe('100');
    })

    test('Debe de incrementar con el boton +1',()=>{
        const wrapper = shallow(<CounterApp/>);

        // let valorActual = wrapper.find('p').text();
        // console.log(valorActual);

        wrapper.find('button').at(0).simulate('click');
        const valorClick = wrapper.find('p').text();
        console.log(valorClick);
        expect(valorClick).toBe('101');
    });

    test('Debe de decrementar con el boton -1',()=>{
        const wrapper = shallow(<CounterApp/>);

        wrapper.find('button').at(2).simulate('click');
        const valorClick = wrapper.find('p').text();
        console.log(valorClick);
        expect(valorClick).toBe('99');
    });

    test('Debe de colocar el valor por defecto con el btn reset' ,()=>{
        const wrapper = shallow(<CounterApp count={ 105 }/> );

        // Visualizar valor actual del counter
        const counterText = wrapper.find('p').text().trim();
        console.log(counterText);

        // Incrementamos el valor
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        const counterIncrement = wrapper.find('p').text().trim();
        console.log(counterIncrement);

        // Reset del valor
        wrapper.find('button').at(1).simulate('click');
        const counterReset = wrapper.find('p').text().trim();
        console.log(counterReset);

        expect(counterReset).toBe('105');


    });


});
