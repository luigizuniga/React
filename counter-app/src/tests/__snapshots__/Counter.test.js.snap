// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <CounterApp/> Debe de mostrar <CounterApp /> correctamente 1`] = `
<Fragment>
  <h2>
    Bienvenido a CounterApp
  </h2>
  <p>
    100
  </p>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
